src/test/java/io/nats/client/impl/ReconnectTests.java

```diff
@@ -165,16 +145,13 @@ public void testSubscribeDuringReconnect() throws Exception {
             sub = nc.subscribe("subsubject");
                 
             final NatsConnection nnc = nc;
-            Dispatcher d = nc.createDispatcher((msg) -> {
-                nnc.publish(msg.getReplyTo(), msg.getData());
-            });
+            Dispatcher d = nc.createDispatcher((msg) -> nnc.publish(msg.getReplyTo(), msg.getData()));
             d.subscribe("dispatchSubject");
 
             handler.prepForStatusChange(Events.RECONNECTED);
 
             try (NatsTestServer ts = new NatsTestServer(port, false)) {
-                handler.waitForStatusChange(400, TimeUnit.MILLISECONDS);
-                assertTrue(Connection.Status.CONNECTED == nc.getStatus(), "Connected Status");
+                waitThenAssertConnected(nc, handler);
```

Comment:
not enough time to ensure connect. use macro since this not the point of the test